lib/_processQueue.ts

```diff
@@ -5,27 +5,40 @@
  * instead of putting them to the queue
  * @return {[type]} [description]
  */
-export function processQueue() {
+import { isFunction } from "./utils";
+
+export function processQueue(globalObject) {
   // Set pointer which allows renaming of the script
-  const pointer = window["AlgoliaAnalyticsObject"] as any;
+  const pointer = globalObject["AlgoliaAnalyticsObject"] as any;
 
   // Check if there is a queue
   if (pointer) {
-    const queue = window[pointer].queue || [];
+    const queue = globalObject[pointer].queue || [];
 
     // Loop queue and execute functions in the queue
-    queue.forEach((fn: string[]) => {
-      const functionName = fn[0];
-      const functionArguments = fn[1];
+    queue.forEach((fn: any[]) => {
+      const functionName: string = fn[0];
+      const functionArguments: any = fn[1];
+      const functionCallback: (err: any, res: any) => void = fn[2];
 
-      if (functionName && typeof (this as any)[functionName] === "function") {
-        this[functionName](functionArguments);
+      if (functionName && isFunction((this as any)[functionName])) {
```

Comment:
Just by curiosity, is there a reason why skip the function here but not once the queue is proceed?

Reply:
It's one of the inconsistencies I was thinking of in my answer to François https://github.com/algolia/search-insights.js/pull/107#discussion_r274337122. 
It's been like that for a while. 
I was thinking in a later iteration I'd factorize that code into a single function `processQueueItem` and handle their all theses inconsistencies.